### ISSUE TITLE
Using with versions ≥ `2.1.2.04` may get your account permanently banned.

### ISSUE THREAD

**Issue by FIM43-Redeye** [gucci-on-fleek/lockdown-browser#65]
![image](https://github.com/gucci-on-fleek/lockdown-browser/assets/14264359/c68d4663-d11b-4a0e-b096-4fccf677d87e)
No longer operational. No idea how or what was done.

**Comment by kiwi08522** [comment]
Thanks for updating. Do you know what version your LDB installer is? I'm wondering if it is only patched in the newest versions or if Respondus automatically patches each client.

**Comment by FIM43-Redeye** [comment]
2-1-2-04, but it sort of... updates itself on first run after installing?

**Comment by whatiswrongwithdota** [comment]
got permanently banned while i logged in to my account! guess d2 doesnt bans for fun now?

**Comment by kiwi08522** [comment]
@FIM43-Redeye Hmm. I have 2-1-2-02, and there's this:

Still working on my setup.  
![image](https://github.com/gucci-on-fleek/lockdown-browser/assets/143338638/1b21419c-c2b6-4df5-8e5e-3543eb24bb6c)

_Originally posted by @carrie-leigh-221 in https://github.com/gucci-on-fleek/lockdown-browser/discussions/53#discussioncomment-8909410_

Pardon me, I'm kind of dumb.  Does this answer your question? 
![image](https://github.com/gucci-on-fleek/lockdown-browser/assets/143338638/8c4cf44a-87be-4259-85bd-91f9b0a4864a)

_Originally posted by @carrie-leigh-221 in https://github.com/gucci-on-fleek/lockdown-browser/discussions/53#discussioncomment-8914754_

I'm wondering if it stopped working on 2-1-2-04?

**Comment by kiwi08522** [comment]
@FIM43-Redeye I don't think it updates itself, it just uses the installer to install that version of Lockdown to the VM. I'm using a 2.1.2.02 installer and after it installs and opens Lockdown, when I check what version I am running, it says 2.1.2.02.
However, this still doesn't mean that the version numbers mean nothing and Lockdown pushes these patches to every Lockdown client, which is definitely possible considering Lockdown requires internet connectivity. But since it is working for @carrie-leigh-221, I think anyone on 2.1.2.02 or earlier is in the clear.

**Comment by mk6398** [comment]
Running "LockDownBrowser-2-1-2-04-364713981.exe" and works fine for me, however, if it updates on startup it will detect it but it has been like this for awhile.

**Comment by fri12013** [comment]
hi dose any one knew how do i start fake test to check that is evrething is fine?

**Comment by ghostpanda63** [comment]
I am running into the same issue. 
Did anyone find a fix ?

**Comment by fri12013** [comment]
@ghostpanda63
 dose it Detect when you start the exam or when just opeing the app?

**Comment by ghostpanda63** [comment]
@fri12013 It detected after starting the exam. I didnt have an issue when it was at 2.1.2.02 but when I updated it to 2.1.2.04 this was happening. I also got banned.

**Comment by fri12013** [comment]
@ghostpanda63

did the LDB updates itself when you run it? or you have but the lastet Version of LDB in the runtime_directory 

if it updates itself thats may be whay im Not sure, waiting for your response

**Comment by carrie-leigh-221** [comment]
> Running "LockDownBrowser-2-1-2-04-364713981.exe" and works fine for me, however, if it updates on startup it will detect it but it has been like this for awhile.
> 
> You just need to download the latest version and replace it in your directory.

I downloaded the latest version and replaced it in my directory.  Logged in to take a quiz.  Got to the last step before it let me start the quiz, and got the warning.

![image](https://github.com/gucci-on-fleek/lockdown-browser/assets/143338638/b4ee74e9-cc2e-4de8-aeb1-8e394d0fa99e)

**Comment by kiwi08522** [comment]
@carrie-leigh-221 Did the older version you were on stop working as well?

**Comment by carrie-leigh-221** [comment]
> @carrie-leigh-221 Did the older version you were on stop working as well?

The older version did not stop working, however when the older version is in the runtime directory folder, it auto updates to the newest one.  Anybody know a way to make it not auto update?

**Comment by ghostpanda63** [comment]
@fri12013 I had the lastest version in the runtime_directory. From their website, it says that they updated their security for 2.1.2.04
![image](https://github.com/gucci-on-fleek/lockdown-browser/assets/72174019/7083be47-aaf3-439d-ba48-89aaa564ee33)

**Comment by gucci-on-fleek** [comment]
Well that's unfortunate. As I mentioned in the [“Project Status” discussion](https://github.com/gucci-on-fleek/lockdown-browser/discussions/53#discussion-6203261), I have neither the time, interest, nor ability to figure out a new bypass method, so we're probably stuck here until/unless someone [submits a PR](https://github.com/gucci-on-fleek/lockdown-browser/pulls) to fix this.

I've pinned this post so that hopefully others can see this warning. Thanks for the update though.

**Comment by gucci-on-fleek** [comment]
It appears that this ban is likely tied to your school account, not your computer as per @bluesangheili from #71:

> I tried to mess around with ldb from two entirely different computers, one warning in the first one, second warning to my other computer. The only way I can make sense out of this is they are targeting your login credentials. OEM ldbs are not affected by this. 

So I'd recommend being very careful if you get this message, because if your entire school account gets banned there's going to be no way to fix things short of a _very awkward_ conversation with your school administration.

**Comment by RowdyRondyRonnie** [comment]
Is there a way to uninstall this? I tried it once and it worked but after the update it doens't work anymore. The repo from my pc and uninstalled sandbox as well but is there anything else I'm missing?

**Comment by gucci-on-fleek** [comment]
@RowdyRondyRonnie

> Is there a way to uninstall this? I tried it once and it worked but after the update it doens't work anymore. The repo from my pc and uninstalled sandbox as well but is there anything else I'm missing?

To “uninstall” this tool itself, all that you need to do is delete the folder that you cloned it to.

For the rest, it depends on what you had on your computer before. For example, before I even created this repo, I had Git, the Visual Studio C++ Tools, and the _Windows Sandbox_ already installed since they're all generally useful for programming on Windows. So if I were to uninstall this tool, I'd still want to keep all of those.

If you have no plans on ever doing any programming in general, then you can uninstall Git. If you have no plans on ever doing any C/C++ programming, then you can uninstall the Visual Studio Build Tools. I'd recommend keeping the _Sandbox_ installed though—it's builtin to Windows, so it automatically updates and takes hardly any storage, and it's really useful for running untrusted software in general. It's up to you though.

**Comment by bumfany** [comment]
No issues on my end with 2-1-2-06. Tried a practice test and it was fine. Did you test it before the exam on a practice exam, or was this out of the blue? @ghostpanda63

**Comment by ghostpanda63** [comment]
It was completely out of the blue

**Comment by bumfany** [comment]
I'm not sure what would trigger it for you. I'm unaware of what happens on the backend or over the network; it might be machine or OS specific, or possibly implementation-specific. In my opinion, the best way to handle ldb or others in the future is to use the old-school method of directly patching them with IDA.

**Comment by onthegrid007** [comment]
https://github.com/LloydLabs/wsb-detect/ is probably how it is being detected, if anyone has any free time please look into this a start a patch

**Comment by Voidless7125** [comment]
I might have something to add to this. The clipboard redirection can block ctrl and alt and see it on host when lockdown is open, so just turn it off (its also easy to see that you copied and pasted). I'll make a pr for editing the VM opener in the meantime.

**Comment by NicholasFlamy** [comment]
> I might have something to add to this. The clipboard redirection can block ctrl and alt and see it on host when lockdown is open, so just turn it off (its also easy to see that you copied and pasted). I'll make a pr for editing the VM opener in the meantime.

Man, I stumbled upon this thread at the perfect time. Gonna check out your fork.

I haven't done anything other than open the lockdown browser using an rldb link that I copied from the JS log in F12 Dev Menu and pasted into explorer in the sandbox. Hit start quiz but then it didn't detect my webcam even though I ran `Sandbox-with-Microphone-Camera.wsb`.

**Comment by Voidless7125** [comment]
> > I might have something to add to this. The clipboard redirection can block ctrl and alt and see it on host when lockdown is open, so just turn it off (its also easy to see that you copied and pasted). I'll make a pr for editing the VM opener in the meantime.
> 
> Man, I stumbled upon this thread at the perfect time. Gonna check out your fork.
> 
> I haven't done anything other than open the lockdown browser using an rldb link that I copied from the JS log in F12 Dev Menu and pasted into explorer in the sandbox. Hit start quiz but then it didn't detect my webcam even though I ran `Sandbox-with-Microphone-Camera.wsb`.
> 
> Edit: I also started my OBS Virtual Camera just to check if the issue was that I'm using my GoPro as a webcam.
> 
> Edit 2: Also, I'm not on 24H2 I'm on 23H2 so I don't think that's the webcam issue.
> 
> Edit 3: The GoPro webcam seems to be the problem since the Windows Camera app on the host doesn't detect it.

Thanks for checking it out! There was a note about the camera app and other apps missing from 24H2, and (although not noted in the doc) it seems to have removed the implementation for the time being. I tested this on 23H2, and the cam worked.

![Image](https://github.com/user-attachments/assets/a67bd1ea-c367-4a03-bffe-92f048bff354)

**Comment by NicholasFlamy** [comment]
> > > I might have something to add to this. The clipboard redirection can block ctrl and alt and see it on host when lockdown is open, so just turn it off (its also easy to see that you copied and pasted). I'll make a pr for editing the VM opener in the meantime.
> > 
> > 
> > Man, I stumbled upon this thread at the perfect time. Gonna check out your fork.
> > I haven't done anything other than open the lockdown browser using an rldb link that I copied from the JS log in F12 Dev Menu and pasted into explorer in the sandbox. Hit start quiz but then it didn't detect my webcam even though I ran `Sandbox-with-Microphone-Camera.wsb`.
> > Edit: I also started my OBS Virtual Camera just to check if the issue was that I'm using my GoPro as a webcam.
> > Edit 2: Also, I'm not on 24H2 I'm on 23H2 so I don't think that's the webcam issue.
> > Edit 3: The GoPro webcam seems to be the problem since the Windows Camera app on the host doesn't detect it.
> 
> Thanks for checking it out! There was a note about the camera app and other apps missing from 24H2, and (although not noted in the doc) it seems to have removed the implementation for the time being. I tested this on 23H2, and the cam worked.
> 
> ![Image](https://github.com/user-attachments/assets/a67bd1ea-c367-4a03-bffe-92f048bff354)

Yeah, I mentioned that in my edits. It seems like the issue is Windows Sandbox does not pass through software-based directshow webcams like OBS Virtual Camera or GoPro Webcam (app that connects to physical GoPro over USB).
